packages/big-design/src/components/FileUploader/FileUploader.tsx

```diff
@@ -0,0 +1,229 @@
+
+  useInputErrors(id, error);
+
+  const validationErrors = useMemo(() => validateFiles(files, validators), [files, validators]);
+
+  useEffect(() => {
+    getImagesPreview(files).then((previews) => {
+      setImagePreview(previews);
+    });
+  }, [files]);
+
+  useEffect(() => {
+    if (onFilesError && validationErrors.length > 0) {
+      onFilesError(validationErrors);
+    }
+  }, [files, onFilesError, validationErrors, validators]);
+
+  const handleFilesChange = useCallback(
+    (newFiles: FileList | null) => {
+      const filesList: File[] = Array.from(newFiles || []);
+
+      if (!multiple) {
+        return onFilesChange([filesList[0]]);
+      }
+
+      onFilesChange([...files, ...filesList]);
+    },
+    [files, multiple, onFilesChange],
+  );
+
+  const handleFileRemove = useCallback(
+    (removedName: string, removedIdx: number) => {
+      const filteredFiles = files.filter(
+        ({ name }, idx) => !(removedName === name && removedIdx === idx),
+      );
+
+      onFilesChange(filteredFiles);
+    },
+    [files, onFilesChange],
+  );
+
+  const renderedLabel = useMemo(() => {
+    if (!label) {
+      return null;
+    }
+
+    if (typeof label === 'string') {
+      return (
+        <FormControlLabel htmlFor={id} localization={localization} renderOptional={!props.required}>
+          {label}
+        </FormControlLabel>
+      );
+    }
+
+    if (
+      isValidElement<LabelHTMLAttributes<HTMLLabelElement>>(label) &&
+      label.type === FormControlLabel
+    ) {
+      return cloneElement(label, {
+        id: labelId,
+        htmlFor: id,
+      });
+    }
+
+    warning('label must be either a string or a FormControlLabel component.');
+  }, [id, label, labelId, localization, props.required]);
+
+  const renderedDescription = useMemo(() => {
+    if (!description) {
+      return null;
+    }
+
+    if (typeof description === 'string') {
+      return <FormControlDescription>{description}</FormControlDescription>;
+    }
+
+    if (isValidElement(description) && description.type === FormControlDescription) {
+      return description;
+    }
+
+    warning('description must be either a string or a FormControlDescription component.');
+  }, [description]);
+
+  const renderedFiles = useMemo(() => {
+    if (isPreviewHidden) {
+      return null;
+    }
+
+    const filesList = files.map((file, idx) => {
+      const isFileInvalid = validationErrors.some(({ fileIdx }) => fileIdx === idx);
+      const imageSrc = imagePreview[idx];
+
+      return (
+        <File
+          isSingle={!multiple}
+          isValid={!isFileInvalid}
+          key={idx}
+          name={file.name}
+          onRemove={(name) => handleFileRemove(name, idx)}
+          previewSrc={imageSrc}
+        />
+      );
+    });
+
+    return <div role={multiple ? 'list' : undefined}>{filesList}</div>;
+  }, [files, handleFileRemove, imagePreview, isPreviewHidden, multiple, validationErrors]);
```

Comment:
According to the above comment I'd recommend to separate these to branches(`multiple` / `single`).

If `multiple === true`, it's generally better to use semantic HTML elements like `<ul> and <li>` for lists whenever possible due to their inherent semantic, accessibility, and SEO benefits.

Reply:
Good point, I have updated it according to your comment.
thanks 😊